Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM rust:1.53 as build
+FROM rust:1.58.1 as build
```

Comment:
I think this is to match with the minimum version mentioned [here](https://github.com/cloudflare/quiche/blob/master/README.md?plain=1#L249)
If minimum version is changed we need to bump README too.

Reply:
It would be nice to actually support formal MSRV now too

Reply:
I tried some rust version and currently the practical minimum version is `1.54.0` because of https://github.com/paupino/rust-decimal#minimum-rust-compiler-version (one of the indirect dependency)

Ideally `rust-version` need to be used but its minimum is `1.56.0` https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field

Reply:
`rust-decimal` is added in the dependency list because of newly added `sfv`.